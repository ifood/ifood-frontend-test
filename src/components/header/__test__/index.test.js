import React from 'react'
import { useSelector } from 'react-redux'
import '@testing-library/jest-dom'
import { render, cleanup, fireEvent } from '@testing-library/react'

import { Header, HeaderProvider } from '../index'

afterEach(cleanup)

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => jest.fn(),
}))

describe('<Header />', () => {
  beforeEach(() => {
    useSelector.mockImplementation((cb) =>
      cb({
        filter: {
          filters: [],
          currentFilters: {},
          hide: true,
        },
        playlist: {
          playlists: [],
        },
      })
    )
  })

  afterEach(() => {
    useSelector.mockClear()
  })

  it('Snapshot testing Playlist', () => {
    const { asFragment } = render(<HeaderProvider />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('Should be able to call onClick prop', () => {
    const mockedOnclick = jest.fn()
    // eslint-disable-next-line prettier/prettier
    const { getByTestId } = render(<Header onClick={mockedOnclick} hide={false} />)

    const headerButton = getByTestId('header-button')

    expect(mockedOnclick).toHaveBeenCalledTimes(0)

    fireEvent.click(headerButton)
    expect(mockedOnclick).toHaveBeenCalledTimes(1)
  })
})
