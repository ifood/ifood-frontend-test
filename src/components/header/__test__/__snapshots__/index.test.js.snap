// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Snapshot testing Playlist 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnylx cBvgHG"
  >
    <img
      class="sc-gtssRu eYzPLv"
      data-testid="header-button"
      src="hamburguer.svg"
    />
    <div
      class="sc-hKFyIo kpVxPh"
    >
      <input
        class="sc-eCApGN iNUwnw"
        placeholder="Search"
      />
    </div>
    <div
      class="sc-dlnjPT cyyyKE"
    />
  </div>
</DocumentFragment>
`;
